lib/Runtime/Debug/DebugContext.h

```diff
@@ -51,11 +51,22 @@ namespace Js
         void SetHostDebugContext(HostDebugContext * hostDebugContext);
 
         void SetDebuggerMode(DebuggerMode mode);
+
         bool IsDebugContextInNonDebugMode() const { return this->debuggerMode == DebuggerMode::NotDebugging; }
         bool IsDebugContextInDebugMode() const { return this->debuggerMode == DebuggerMode::Debugging; }
         bool IsDebugContextInSourceRundownMode() const { return this->debuggerMode == DebuggerMode::SourceRundown; }
         bool IsDebugContextInSourceRundownOrDebugMode() const { return IsDebugContextInSourceRundownMode() || IsDebugContextInDebugMode(); }
 
+        bool IsDebuggerRecording() const { return this->IsDebugContextInDebugMode() && this->isDebuggerRecording; }
```

Comment:
These can't be asserts, but we can remove them to simplify the logic. Besides `AutoInitLibraryCodeScope` save/restore of the `isDebuggerRecording` flag, we only use IsDebuggerRecording in a code path where we already know `IsDebugContextInDebugMode()==true`.



---
In reply to: [130155025](https://github.com/Microsoft/ChakraCore/pull/3447#discussion_r130155025) [](ancestors = 130155025)